tests/unit/test_gate_requirements.py

```
@@ -659,3 +659,63 @@ def get_check(self, name, cfg):
+
+class TestRequiredDepsManifest:
+    """The sm doctor --required-deps emitter (feeds the v2 Action)."""
+
+    def test_aggregate_dedups_by_tool_name(self, monkeypatch):
+        from unittest.mock import MagicMock
+
+        from slopmop.checks import tool_inventory
+
+        black = Requirement(kind="python", name="black", version="26.5.1")
+
+        def gate(*reqs):
```

Ruff (0.15.17)

[warning] Missing return type annotation for private function `gate`

(ANN202)